numba_cuda/numba/cuda/cudadrv/nvvm.py

```diff
@@ -369,37 +369,40 @@ def get_log(self):
     (9, 0),
     (10, 0),
     (10, 1),
+    (10, 3),
     (12, 0),
+    (12, 1),
 )
 
 
 # Maps CTK version -> (min supported cc, max supported cc) inclusive
 _CUDA_CC_MIN_MAX_SUPPORT = {
-    (11, 1): ((3, 5), (8, 0)),
-    (11, 2): ((3, 5), (8, 6)),
-    (11, 3): ((3, 5), (8, 6)),
-    (11, 4): ((3, 5), (8, 7)),
-    (11, 5): ((3, 5), (8, 7)),
-    (11, 6): ((3, 5), (8, 7)),
-    (11, 7): ((3, 5), (8, 7)),
-    (11, 8): ((3, 5), (9, 0)),
-    (12, 0): ((5, 0), (9, 0)),
-    (12, 1): ((5, 0), (9, 0)),
-    (12, 2): ((5, 0), (9, 0)),
-    (12, 3): ((5, 0), (9, 0)),
-    (12, 4): ((5, 0), (9, 0)),
-    (12, 5): ((5, 0), (9, 0)),
-    (12, 6): ((5, 0), (9, 0)),
-    (12, 8): ((5, 0), (12, 0)),
+    (11, 1): [((3, 5), (8, 0)),],
+    (11, 2): [((3, 5), (8, 6)),],
+    (11, 3): [((3, 5), (8, 6)),],
+    (11, 4): [((3, 5), (8, 7)),],
+    (11, 5): [((3, 5), (8, 7)),],
+    (11, 6): [((3, 5), (8, 7)),],
+    (11, 7): [((3, 5), (8, 7)),],
+    (11, 8): [((3, 5), (9, 0)),],
+    (12, 0): [((5, 0), (9, 0)),],
+    (12, 1): [((5, 0), (9, 0)),],
+    (12, 2): [((5, 0), (9, 0)),],
+    (12, 3): [((5, 0), (9, 0)),],
+    (12, 4): [((5, 0), (9, 0)),],
+    (12, 5): [((5, 0), (9, 0)),],
+    (12, 6): [((5, 0), (9, 0)),],
+    (12, 8): [((5, 0), (10, 2)), ((12, 0), (12, 0)),],
+    (12, 9): [((5, 0), (12, 1)),],
 }
 
 
 def ccs_supported_by_ctk(ctk_version):
     try:
         # For supported versions, we look up the range of supported CCs
-        min_cc, max_cc = _CUDA_CC_MIN_MAX_SUPPORT[ctk_version]
         return tuple(
-            [cc for cc in COMPUTE_CAPABILITIES if min_cc <= cc <= max_cc]
+            [cc for cc in COMPUTE_CAPABILITIES if min_cc <= cc <= max_cc
+             for min_cc, max_cc in _CUDA_CC_MIN_MAX_SUPPORT[ctk_version]]
         )
```

Comment:
To accomplish the list of ranges approach, only a slight adjustment in the `list`-comprehension gives us the desired behavior

Reply:
JFYI needed to flip the `for`-loops below: https://github.com/NVIDIA/numba-cuda/pull/273/commits/6211ce5b38ce53ea5464ce1322cdbcb82a20d1e5

Otherwise the idea is the same